web/__init__.py

```diff
@@ -34,30 +64,58 @@
 
 @app.before_first_request
 def startup():
+    """
+    Registers required services on server start
+    """
     app.svc.register("pppp", web.service.pppp.PPPPService())
     app.svc.register("videoqueue", web.service.video.VideoQueue())
     app.svc.register("mqttqueue", web.service.mqtt.MqttQueue())
     app.svc.register("filetransfer", web.service.filetransfer.FileTransferService())
 
 
+def restart():
+    """
+    Restarts the registered services to apply new changes
+    """
+    try:
+        for name in app.svc:
+            app.svc.svcs[name].restart()
+            time.sleep(2)
+    except Exception as err:
+        log.exception(err)
+        return Exception(err)
```

Comment:
Are we returning exceptions now?

Since the try-except is on the outside of the loop, this changes almost nothing, compared to not having the loop.

Reply:
it is to return the exception back to the function that called it as an exception instead of a return message.

Reply:
Yes sorry, it should be `raise` not `return` that is my own mixup, I have now eliminated the try-except here and moved the log functionality into the reload command.

Reply:
Doing that actually causes the exception encountered when the `filetransfer` service fails to stop to pass to the front-end and emit a failure in reloading the services. if you wait a few seconds then refresh the page the video is then reloaded

Reply:
Great - looks much simpler now :)